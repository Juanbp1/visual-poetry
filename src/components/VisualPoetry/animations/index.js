/**
 * @module animations
 * @description Biblioteca de animaciones para crear efectos visuales dinámicos en la aplicación.
 * Proporciona una serie de componentes de animación reutilizables para mejorar la experiencia del usuario.
 *
 * @see {@link module:animations/AnimatedHeader|AnimatedHeader} - Componente para animar el encabezado con efectos personalizados.
 * @see {@link module:animations/FadeIn|FadeIn} - Componente crea una animación de deslizamiento lateral .
 * @see {@link module:animations/Pulse|Pulse} - Componente para aplicar un efecto de pulsación a un elemento.
 * @see {@link module:animations/SmoothTransition|SmoothTransition} - Componente para aplicar transiciones suaves entre estados.
 * @see {@link module:animations/Spin|Spin} - Componente para aplicar una animación de rotación a un elemento.
 */
export { default as AnimatedHeader } from './AnimatedHeader';
export { default as FadeIn } from './FadeIn';
export { default as Pulse } from './Pulse';
export { default as SmoothTransition } from './SmoothTransition';
export { default as Spin } from './Spin';
