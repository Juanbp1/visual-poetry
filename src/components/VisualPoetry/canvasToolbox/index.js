/**
 * @module canvasToolbox
 * @description Biblioteca de herramientas para la edición y manipulación de lienzos. Incluye varios componentes
 * que facilitan la interacción con el lienzo y el control del contenido visual.
 *
 * @see {@link module:canvasToolbox/CanvasDownloader|CanvasDownloader} - Componente para descargar el contenido del lienzo.
 * @see {@link module:canvasToolbox/RenderFormatButtons|RenderFormatButtons} - Componente para mostrar los botones con los formatos a exportar.
 * @see {@link module:canvasToolbox/editor/CanvasTextEditor|CanvasTextEditor} - Componente para editar texto dentro del lienzo.
 * @see {@link module:canvasToolbox/editor/DesktopEditor|DesktopEditor} - Componente para la versión de escritorio del editor de texto.
 * @see {@link module:canvasToolbox/editor/MobileEditor|MobileEditor} - Componente para la versión móvil del editor de texto.
 * @see {@link module:canvasToolbox/SelectionTool|SelectionTool} - componente que proporciona una interfaz para controlar el modo de selección de la imagen.
 * @see {@link module:canvasToolbox/SelectionButton|SelectionButton} - Componente auxiliar para representar un botón de selección
 * @see {@link module:canvasToolbox/ColorPickerControl|ColorPickerControl} - Componente para seleccionar colores para el lienzo.
 * @see {@link module:canvasToolbox/SelectDropdown|SelectDropdown} - Componente de selección desplegable para opciones.
 * @see {@link module:canvasToolbox/Slider|Slider} - Componente de deslizador para ajustar valores.
 */

export { default as CanvasDownloader } from './CanvasDownloader';
export { default as RenderFormatButtons } from './RenderFormatButtons';
export { default as CanvasTextEditor } from './editor/CanvasTextEditor';
export { default as DesktopEditor } from './editor/DesktopEditor';
export { default as MobileEditor } from './editor/MobileEditor';
export { default as SelectionTool } from './SelectionTool';
export { default as SelectionButton } from './SelectionButton';
export { default as ColorPickerControl } from './ColorPickerControl';
export { default as SelectDropdown } from './SelectDropdown';
export { default as Slider } from './Slider';
