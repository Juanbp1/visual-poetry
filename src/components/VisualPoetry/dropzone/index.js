/**
 * @module dropzone
 * @description Biblioteca de componentes para Dropzone. Proporciona una serie de componentes reutilizables
 * para la implementación de una zona de carga de archivos interactiva y personalizada.
 *
 * @see {@link module:dropzone/Dropzone|Dropzone} - Componente que maneja la zona de arrastre y carga de imágenes.
 * @see {@link module:dropzone/DropzoneContent|DropzoneContent} -  Componente que muestra el contenido del área de arrastre (dropzone).
 * @see {@link module:dropzone/DropzoneButtons|DropzoneButtons} - Componente que renderiza los botones para elegir una imagen desde la galería o subir una imagen desde el dispositivo.
 * @see {@link module:dropzone/DropzoneErrorMessage|DropzoneErrorMessage} - Componente para mostrar mensajes de error relacionados con la carga de archivos.
 */
export { default as Dropzone } from './Dropzone';
export { default as DropzoneContent } from './DropzoneContent';
export { default as DropzoneButtons } from './DropzoneButtons';
export { default as DropzoneErrorMessage } from './DropzoneErrorMessage';
