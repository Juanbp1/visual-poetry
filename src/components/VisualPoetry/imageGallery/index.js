/**
 * @module imageGallery
 * @description Biblioteca de componentes relacionados con la galería de imágenes. Proporciona una serie de componentes reutilizables
 * para la visualización y gestión de imágenes dentro de una galería.
 * 
 * @see {@link module:imageGallery/ImageGalleryContainer|ImageGalleryContainer} - Componente que muestra una galería de imágenes con funcionalidades específicas.
 * @see {@link module:imageGallery/ImageList|ImageList} - Componente que lista imágenes en un formato de lista.
 * @see {@link module:imageGallery/Images|Images} - Componente que maneja la visualización de imágenes individuales.
 */

export { default as ImageGalleryContainer } from './ImageGalleryContainer';
export { default as ImageList } from './ImageList';
export { default as Images } from './Images';