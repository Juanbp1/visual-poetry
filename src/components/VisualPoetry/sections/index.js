/**
 * @module sections
 * @description Biblioteca de componentes reutilizables. Proporciona una serie de secciones del Home.
 *
 * @see {@link module:sections/DropzoneContainer|DropzoneContainer} - Componente que maneja áreas de carga mediante arrastrar y soltar.
 * @see {@link module:sections/ExampleGallery|ExampleGallery} - Componente que muestra una galería de ejemplos.
 * @see {@link module:sections/FAQSection|FAQSection} -  componente muestra una sección de preguntas frecuentes
 * @see {@link module:sections/Faqs|Faqs} - Componente renderiza una lista de preguntas frecuentes
 * @see {@link module:sections/FrontPages|FrontPages} - Componente que maneja las páginas principales de la aplicación.
 * @see {@link module:sections/QuickStart|QuickStart} - Componente que proporciona una guía de inicio rápido.
 * @see {@link module:sections/Main|Main} - Componente principal de la aplicación.
 * @see {@link module:sections/videoPlayer/VideoPlayer|VideoPlayer} - Componente para la reproducción de video
 * @see {@link module:sections/videoPlayer/VideoPlayer/BrowserVideoPlayer|BrowserVideoPlayer} - Componente para la reproducción de video en dispositivo escritorio.
 * @see {@link module:sections/videoPlayer/VideoPlayer/MobileVideoPlayer|MobileVideoPlayer} - Componente para la reproducción de video en dispositivos móviles.
 * @see {@link module:sections/Steps/Steps|Steps} - Componente que muestra una serie de pasos o tutoriales.
 * @see {@link module:sections/Steps/Step|Step} - Componente que muestra una serie de pasos o tutoriales.
 * @see {@link module:sections/Checks|Checks} - Componente que muestra una lista de puntos clave con íconos de verificación.
 */

export { default as DropzoneContainer } from './DropzoneContainer';
export { default as ExampleGallery } from './ExampleGallery';
export { default as FAQSection } from './FAQSection';
export { default as Faqs } from './Faqs';
export { default as FrontPages } from './FrontPages';
export { default as QuickStart } from './QuickStart';
export { default as Main } from './Main';
export { default as VideoPlayer } from './videoPlayer/VideoPlayer';
export { default as BrowserVideoPlayer } from './videoPlayer/BrowserVideoPlayer';
export { default as MobileVideoPlayer } from './videoPlayer/MobileVideoPlayer';
export { default as Steps } from './steps/Steps';
export { default as Step } from './steps/Step';
export { default as checks } from './checks/Checks';
