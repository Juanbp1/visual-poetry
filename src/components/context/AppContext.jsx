/**
 * @fileoverview Contexto y Reducer para la aplicación.
 * @module AppContext
 */
import React, { createContext, useContext, useReducer } from 'react';
import actionTypes from '../constants/actionTypes';
import { TEXT_STRINGS } from '../constants/uiConstants';
import actionsFactory from '../constants/actionsFactory';
import defineActions from '../constants/actions';

/**
 * @typedef {Object} AppState
 * @property {Image|null} image - La imagen actual.
 * @property {ImageBorder|null} imageBorder - El borde de la imagen.
 * @property {boolean} imageIsInverted - Indica si la imagen está invertida.
 * @property {Array<ImageGalleryItem>} imageGallery - La galería de imágenes.
 * @property {Object|null} imageSelected - La imagen seleccionada.
 * @property {boolean} imageIsLoaded - Indica si la imagen ha sido cargada.
 * @property {boolean} imageIsProcessed - Indica si la imagen ha sido procesada.
 * @property {boolean} inputIsSetted - Indica si la entrada está configurada.
 * @property {Object} imageBorderPixels - Los píxeles del borde de la imagen.
 * @property {Object} imageData - Datos de la imagen.
 * @property {Object} inputPositionCoordinates - Coordenadas de posición de entrada.
 * @property {string} inputData - Datos de entrada.
 * @property {Object} mainContext - Contexto principal del canvas.
 * @property {Object} secondContext - Segundo contexto del canvas.
 * @property {Object} canvasSize - Tamaño del canvas.
 * @property {Object} styleFont - Estilo de fuente actual.
 * @property {string|null} errorUploadImage - Error al subir la imagen.
 * @property {number} figureThreshold - Umbral para figuras.
 * @property {boolean} loading - Estado de carga.
 * @property {boolean} showGallery - Indica si se muestra la galería.
 * @property {boolean} showGallerySecond - Indica si se muestra la segunda galería.
 * @property {boolean} showGalleryThird - Indica si se muestra la tercera galería.
 */

/**
 * @type {React.Context<{state: AppState, dispatch: React.Dispatch<any>, actions: Object}>}
 */
const AppContext = createContext();

/**
 * Estado inicial del contexto de la aplicación.
 * @type {AppState}
 */
const initialState = {
  image: null,
  imageBorder: null,
  imageIsInverted: false,
  imageGallery: [],
  imageSelected: null,
  imageIsLoaded: false,
  imageIsProcessed: false,
  inputIsSetted: false,
  imageBorderPixels: {},
  imageData: {},
  inputPositionCoordinates: { start: [], end: [] },
  inputData: TEXT_STRINGS.INITIAL_TEXT,
  mainContext: null,
  secondContext: null,
  canvasSize: { canvasWidth: 0, canvasHeight: 0 },
  styleFont: {
    color: '#000000',
    background: '#FF000000',
    font: 'Arial',
    size: 8,
  },
  errorUploadImage: null,
  figureThreshold: 128,
  loading: false,
  showGallery: false,
  showGallerySecond: false,
  showGalleryThird: false,
};

const {
  CHECK_IMAGE_LOADED,
  CHECK_IMAGE_PROCESSED,
  CHECK_INPUT_SETTED,
  OBTAIN_CANVAS_SIZE,
  OBTAIN_IMAGE,
  OBTAIN_IMAGE_BORDER,
  OBTAIN_IMAGE_BORDER_PIXELS,
  OBTAIN_IMAGE_GALLERY,
  OBTAIN_IMAGE_SELECTED,
  OBTAIN_INPUT_POSITION_COORDINATES,
  OBTAIN_MAIN_CONTEXT,
  OBTAIN_SECOND_CONTEXT,
  RESET,
  TOGGLE_INVERT_IMAGE,
  UPDATE_ERROR_UPLOAD_IMAGE,
  UPDATE_FIGURE_THRESHOLD,
  UPDATE_GALLERY_TOGGLE,
  UPDATE_GALLERY_TOGGLE2,
  UPDATE_GALLERY_TOGGLE3,
  UPDATE_IMAGE_GALLERY,
  UPDATE_INPUT_DATA,
  UPDATE_LOADING,
  UPDATE_STYLE_FONT,
} = actionTypes;

/**
 * Función reductora para manejar acciones en el estado de la aplicación.
 * @param {AppState} state - El estado actual de la aplicación.
 * @param {Object} action - La acción que se va a procesar.
 * @param {string} action.type - El tipo de acción que indica qué operación realizar.
 * @param {any} [action.payload] - Los datos asociados a la acción, si los hay.
 * @returns {AppState} - El nuevo estado de la aplicación después de procesar la acción.
 */
const reducer = (state, action) => {
  switch (action.type) {
    case CHECK_IMAGE_LOADED:
      return { ...state, imageIsLoaded: true };

    case CHECK_IMAGE_PROCESSED:
      return { ...state, imageIsProcessed: true };

    case CHECK_INPUT_SETTED:
      return { ...state, inputIsSetted: true };

    case OBTAIN_CANVAS_SIZE:
      return { ...state, canvasSize: action.payload };

    case OBTAIN_IMAGE:
      return { ...state, image: action.payload };

    case OBTAIN_IMAGE_BORDER:
      return { ...state, imageBorder: action.payload };

    case OBTAIN_IMAGE_BORDER_PIXELS:
      if (action.payload && typeof action.payload.getImageData === 'function') {
        const context = action.payload;
        const imageDataPixels = context.getImageData(
          0,
          0,
          context.canvas.width,
          context.canvas.height
        );
        return { ...state, imageBorderPixels: imageDataPixels };
      }
      return { ...state, imageBorderPixels: {} };
    case OBTAIN_IMAGE_GALLERY:
      return { ...state, imageGallery: action.payload };

    case OBTAIN_INPUT_POSITION_COORDINATES:
      return { ...state, inputPositionCoordinates: action.payload };

    case OBTAIN_MAIN_CONTEXT:
      return { ...state, mainContext: action.payload };

    case OBTAIN_SECOND_CONTEXT:
      return { ...state, secondContext: action.payload };

    case OBTAIN_IMAGE_SELECTED:
      return { ...state, imageSelected: action.payload };

    case RESET:
      return { ...initialState };

    case TOGGLE_INVERT_IMAGE:
      return { ...state, imageIsInverted: action.payload };

    case UPDATE_ERROR_UPLOAD_IMAGE:
      return { ...state, errorUploadImage: action.payload };

    case UPDATE_FIGURE_THRESHOLD:
      return { ...state, figureThreshold: action.payload };

    case UPDATE_GALLERY_TOGGLE:
      return { ...state, showGallery: action.payload };

    case UPDATE_GALLERY_TOGGLE2:
      return { ...state, showGallerySecond: action.payload };

    case UPDATE_GALLERY_TOGGLE3:
      return { ...state, showGalleryThird: action.payload };

    case UPDATE_IMAGE_GALLERY:
      const id = action.payload;
      const imageSelected = state.imageGallery[0].map((img) => {
        return { ...img, selected: img.id === id ? !img.selected : false };
      });
      return { ...state, imageGallery: [imageSelected] };

    case UPDATE_INPUT_DATA:
      return { ...state, inputData: action.payload };

    case UPDATE_LOADING:
      return { ...state, loading: !state.loading };

    case UPDATE_STYLE_FONT:
      const updatedStyleFont = {
        ...state.styleFont,
        ...action.payload,
      };
      return {
        ...state,
        styleFont: updatedStyleFont,
      };

    default:
      return state;
  }
};

/**
 * Proveedor del contexto de la aplicación.
 * @param {Object} props - Las propiedades del componente.
 * @param {React.ReactNode} props.children - Los componentes hijos a envolver.
 * @returns {JSX.Element} - El componente Provider con el contexto.
 */
export const AppProvider = ({ children }) => {
  const [state, dispatch] = useReducer(reducer, initialState);

  const createAction = actionsFactory(dispatch);

  // Definir las acciones utilizando la fábrica de acciones
  const { imageActions, styleActions, inputActions, othersActions } =
    defineActions(createAction);

  const actions = {
    ...imageActions,
    ...styleActions,
    ...inputActions,
    ...othersActions,
  };

  return (
    <AppContext.Provider value={{ state, dispatch, actions }}>
      {children}
    </AppContext.Provider>
  );
};

/**
 * Hook para utilizar el contexto de la aplicación.
 * @returns {Object} - El contexto de la aplicación que incluye el estado, el despachador y las acciones.
 * @throws {Error} - Lanza un error si el hook se usa fuera del proveedor de contexto.
 */
export const useAppContext = () => {
  const context = useContext(AppContext);
  if (!context) {
    throw new Error(
      'useAppContext debe ser utilizado dentro de un AppProvider'
    );
  }
  return context;
};
