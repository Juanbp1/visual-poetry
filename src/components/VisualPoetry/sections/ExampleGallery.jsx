import React, { useEffect, useState } from 'react';
import {
  IMAGE_GALLERY_EXAMPLE_DATA,
  CONTEXTS,
} from '../../constants/uiConstants';
import { ImageList } from '../imageGallery';
import { createImageDataList, scrollToSection } from '../../utils';
import { Button } from '../../common';
import { BsFillArrowUpCircleFill } from 'react-icons/bs';

/**
 * Este componente carga una lista de imágenes de ejemplo utilizando `createImageDataList` y las muestra en una cuadrícula.
 *
 * @module sections/ExampleGallery
 * @example
 * // Ejemplo de uso del componente ExampleGallery
 * import React from 'react';
 * import { ExampleGallery } from '../sections';
 *
 * const Mycomponent = () => {
 *    return (
 *      <ExampleGallery />
 *    );
 *  }
 * export default MyComponent;
 */
const ExampleGallery = () => {
  const [images, setImages] = useState([]);
  useEffect(() => {
    const images = createImageDataList(
      IMAGE_GALLERY_EXAMPLE_DATA,
      CONTEXTS.imagesGalleryExample
    );
    setImages([images]);
  }, []);
  return (
    <section
      className="sections__exampleGalleryContainer exampleGalleryContainer"
      id="exampleGalleryId"
      data-testid="exampleGalleryId"
    >
      <div className="exampleGalleryContainer__exampleGallery exampleGallery">
        <div className="exampleGallery__article">
          <h2
            className="exampleGallery__title"
            data-testid="exampleGalleryTitle"
          >
            Inspírese con Diseños Creativos
          </h2>
          <p className="exampleGallery__text" data-testid="exampleGalleryText">
            Explore una galería de diseños creados con Visual Poetry para
            obtener ideas y descubrir su potencial
          </p>
        </div>
        <div className="exampleGallery__grid">
          <ImageList
            imagesList={images}
            classNameWrapper="exampleGallery__item"
            classNameImg="exampleGallery__img"
            id="exampleImage"
          />
        </div>
        <Button
          className="button__exampleGallery button__withIcon"
          name="Comenzar a crear"
          text="Comenzar a crear"
          Icon={BsFillArrowUpCircleFill}
          iconClassName="button__frontepageArrowIcon"
          handleClick={() => scrollToSection('dropzoneId')}
          title="Comenzar a crear"
          ariaLabel="Comenzar a crear"
          id="exampleGalleryButton"
        />
      </div>
    </section>
  );
};

export default ExampleGallery;
