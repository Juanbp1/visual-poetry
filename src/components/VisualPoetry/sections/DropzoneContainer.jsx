import React from 'react';
import { ReactCompareSlider } from 'react-compare-slider';
import { ImageGalleryContainer } from '../imageGallery';
import { PiArrowFatLineRightFill } from 'react-icons/pi';
import CompareSlider from '../compareImageSlider/CompareSlider';
import { Dropzone, DropzoneErrorMessage } from '../dropzone';
import { useAppContext } from '../../context/AppContext';

/**
 * Este componente maneja la carga de imágenes y las características de comparación. Permite a los usuarios subir una imagen, ver una comparación entre la imagen original y la editada, y visualizar una galería de imágenes.
 *
 * @module sections/DropzoneContainer
 * @example
 * // Ejemplo de uso del componente DropzoneContainer
 * import React from 'react';
 * import { DropzoneContainer } from '../sections'
 *
 * const MyComponent = () => {
 *    return (
 *      <DropzoneContainer />
 *    );
 * };
 * export default MyComponent;
 */
const DropzoneContainer = () => {
  const { state } = useAppContext();
  return (
    <section
      className="sections__dropzoneContainer dropzoneContainer"
      id="dropzoneId"
      data-testid="dropzoneId"
    >
      <div className="dropzoneContainer__dropzoneElement dropzoneElement">
        <div className="dropzoneElement__article">
          <h2 className="dropzoneElement__title" data-testid="dropzoneTitle">
            Transforma tus imágenes en arte textual
          </h2>
          <p className="dropzoneElement__text" data-testid="dropzoneText">
            Sube una imagen y conviértela en una obra de arte única con nuestro
            editor de texto en lienzo.
          </p>
        </div>
        <DropzoneErrorMessage />
        <div className="dropzoneElement__wrapper">
          <div className="dropzoneElement__compareImageContainer">
            <div className="dropzoneElement__compareImage">
              <CompareSlider ReactCompareSlider={ReactCompareSlider} />
            </div>
            <p className="dropzoneElement__compareImageText">
              Imagen original (izquierda) Resultado final (derecha)
            </p>
          </div>
          <PiArrowFatLineRightFill
            className="dropzoneElement__arrowIcon"
            aria-hidden="true"
          />
          <Dropzone imageData={state.imageSelected} />
          <ImageGalleryContainer />
        </div>
      </div>
    </section>
  );
};
export default DropzoneContainer;
