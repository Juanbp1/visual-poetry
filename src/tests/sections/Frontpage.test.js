import React from 'react';
import { cleanup, render, screen } from '@testing-library/react';
import { AppProvider } from '../../components/context/AppContext';
import { mockUseMediaQuery } from '../mockUseMediaQuery';
import {
  BrowserVideoPlayer,
  MobileVideoPlayer,
} from '../../components/VisualPoetry/sections';

import { testScrollToSection } from '../helpers/userHelpers';

import { FrontPages } from '../../components/VisualPoetry/sections';

const renderHomeSection = () => {
  render(
    <AppProvider>
      <FrontPages />
    </AppProvider>
  );
};
describe('La sección frontpage debe funcionar correctamente', () => {
  afterEach(() => {
    cleanup();
    jest.clearAllMocks();
  });

  test('Se debería renderizar el título y la descripción principal', () => {
    renderHomeSection();
    const title = screen.getByTestId('frontpageTitle');
    const text = screen.getByTestId('frontpageText');

    expect(title).toHaveTextContent('Visual Poetry');
    expect(text).toHaveTextContent('Convierte imágenes en arte textual');
  });
  test('El boton "Comenzar a crear" bebería ser representado correctamente', () => {
    renderHomeSection();

    const button = screen.getByTestId('frontpageButton');
    expect(button).toMatchSnapshot();
  });
  test('Debería desplazarse a la sección Dropzone al hacer clic en el botón "Comenzar a crear"', () => {
    testScrollToSection('/', 'frontpageButton', 'dropzoneId', {
      behavior: 'smooth',
      block: 'end',
    });
  });
  test('No debería desplazarse si la sección a la que se intenta ir no es la correcta', () => {
    testScrollToSection('/', 'frontpageButton', 'dropzoneId', {}, false);
  });
  test('El video debería tener las propiedades autoPlay, muted, loop, playsInline y playbackRate', async () => {
    render(
      <AppProvider>
        <BrowserVideoPlayer />
      </AppProvider>
    );
    const videoElement = screen.getByTestId('browserVideoPlayer');
    expect(videoElement).toBeInTheDocument();

    expect(videoElement).toHaveAttribute('autoPlay');
    expect(videoElement).toHaveAttribute('loop');
    expect(videoElement).toHaveAttribute('playsInline');
    expect(videoElement.playbackRate).toBe(1.2);
  });

  test('El video (diseño movíl) debería tener las propiedades autoPlay, muted, loop, playsInline y playbackRate', async () => {
    mockUseMediaQuery({ isMobile: true });
    render(
      <AppProvider>
        <MobileVideoPlayer />
      </AppProvider>
    );
    const videoElement = screen.getByTestId('mobileVideoPlayer');
    expect(videoElement).toBeInTheDocument();

    expect(videoElement).toHaveAttribute('autoPlay');
    expect(videoElement).toHaveAttribute('loop');
    expect(videoElement).toHaveAttribute('playsInline');
    expect(videoElement.playbackRate).toBe(1.2);
  });
});
