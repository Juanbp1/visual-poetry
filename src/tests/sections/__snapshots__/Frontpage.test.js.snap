// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`La sección frontpage debe funcionar correctamente El boton "Comenzar a crear" bebería ser representado correctamente 1`] = `
<button
  aria-label="Comenzar a crear"
  class="button__frontepage button__withIcon"
  data-testid="frontpageButton"
  name="Comenza a crear"
  title="Comenzar a crear"
  type="button"
>
  <svg
    class="button__frontepageArrowIcon"
    fill="currentColor"
    height="1em"
    stroke="currentColor"
    stroke-width="0"
    viewBox="0 0 16 16"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M16 8A8 8 0 1 1 0 8a8 8 0 0 1 16 0zM8.5 4.5a.5.5 0 0 0-1 0v5.793L5.354 8.146a.5.5 0 1 0-.708.708l3 3a.5.5 0 0 0 .708 0l3-3a.5.5 0 0 0-.708-.708L8.5 10.293V4.5z"
    />
  </svg>
  Comenza a crear
</button>
`;
